src/kdf.rs

```diff
@@ -1,26 +1,26 @@
 use crate::CryptoCoreError;
-use generic_array::{ArrayLength, GenericArray};
 use hkdf::Hkdf;
-use sha2::Sha256;
+use sha2::{digest::OutputSizeUser, Sha256};
 
 /// Derive a key of `KEY_LENGTH` bytes using a HMAC-based Extract-and-Expand
 /// Key Derivation Function (HKDF) supplying a `bytes` and some `info` context
 /// `String`. The hash function used is Sha256.
 ///
-/// - `bytes`   : input bytes to hash, should be at least 32-bytes long
+/// - `ikm`     : input key material, should be at least 32-bytes long
 /// - `info`    : some optional additional information to use in the hash
-pub fn hkdf_256<KeyLength: ArrayLength<u8>>(
-    bytes: &[u8],
+pub fn hkdf_256<const LENGTH: usize>(
+    ikm: &[u8],
     info: &[u8],
-) -> Result<GenericArray<u8, KeyLength>, CryptoCoreError> {
-    if bytes.len() < 32 {
-        return Err(CryptoCoreError::InvalidSize(
-            "Input `bytes` size should be at least 32".to_string(),
-        ));
+) -> Result<[u8; LENGTH], CryptoCoreError> {
+    if ikm.len() < <Sha256 as OutputSizeUser>::output_size() {
```

Comment:
I don't get why you need to put a condition on the input bytes. 
For me, a pbkdf is used to create a fixed-size key from a arbitrary length bytes array.

Reply:
I think the aim is to ensure a sufficient level of entropy is contained in the input key material. We are using HKDF, which **is not a PBKDF** in my understanding. The important difference is that there is no slowing down of the KDF operation (which allows PBKDF to derive cryptographically strong outputs from low entropy inputs).

See the [HKDF paper](https://eprint.iacr.org/2010/264) section 9, Password-based KDF.

Reply:
This function is used in the KMS to derive a key from a password. If you want to force the password (bytes array) length in the HKDF function, then we shouldn't use it in the KMS and implement a real PBKDF2 instead. Indeed, we mustn't force the user to select a 32 bytes password. The customer has already probably its own password policy. 

I see 3 solutions. I let you choose which one you'd rather: 
- Do not put that limit in the HKDF function and let the entropy issue to a outer policy ;
- Put that limit but add a PBKDF2 function in crypto core to use in the KMS ;
- Put that limit but add an issue in the KMS (for further) to use a PBKDF2 instead of a HKDF and use variable password length.

Reply:
I removed it entirely.

Reply:
A PBKDF or Argon2 algorithm should be used in KMS to extract keys from passwords. Implementation with sufficient entropy (like CoverCrypt or Findex) should use a SHA algorithm (preferentially SHA-3).

*NOTE*: SHA-3 has a `Xof` mode which allows expending the output to a desired length (it therefore replaces all HKDF functionalities)